### PR TITLE
RA1865 - Fixing reflected XSS in the notes section

### PR DESCRIPTION
**Pull request by varsha5595**
## Description of what I changed

@isears 
Added input sanitization for Notes section in the Appoint Scheduling Form.

## Issue I worked on

This fix ensures iframes are not inserted in the Notes section while scheduling appointments. The bug can be reproduced by following the steps mentioned [here](https://docs.google.com/document/d/1vfltuHZEmgh2wELWww70PG1eUH16qZ_71b7h-xMs7NQ/edit?usp=sharing).

## Link to ticket
https://issues.openmrs.org/browse/RA-1865

## Checklist: I completed these to help reviewers :)

- [x ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by varsha5595** [comment]
@isears 
I looked further into the UiUtils implementation of HTML Encoding and realized that they are directly calling the encodeHtml function. As I encountered problems while using the UiUtils class, I directly used the OWASP standard Encode.forHtml function to handle the XSS issue here. 

I also looked into fixing this at the GSP level, but because the textarea has a two-way binding to a variable using ng-model, I was unsure how to go about it.

File: pom.xml [hunk 1]
```diff
@@ -133,6 +133,12 @@
             <version>2.2</version>
         </dependency>
 
+		<dependency>
+           <groupId>org.owasp.encoder</groupId>
```
Comment (dkayiwa) [inline pom.xml:137]: Doesn't having the same library in multiple modules result into this problem? https://issues.openmrs.org/browse/TRUNK-5344

How about just relying on the one in core? https://github.com/openmrs/openmrs-core/blob/2.0.0/web/pom.xml#L116-L120

Comment (varsha5595) [reply pom.xml]: I have changed the implementation to use the escapeHTML() from the already existing WebUtil class from the core.

**Review by isears — CHANGES_REQUESTED** [review]
In general, pulling escapeHTML from WebUtil is a really nice solution to the importing problem we've been having.

In this specific case, though, it looks like we're double-encoding  at `/openmrs/appointmentschedulingui/manageAppointments.page`:
image

At which pages specifically do these XSS vulnerabilities show up? Appointment dashboard or patient dashboard? Is patching in the corresponding .gsp pages an option?

File: api/src/main/java/org/openmrs/module/appointmentscheduling/AppointmentRequest.java [hunk 2]
```diff
@@ -90,12 +91,16 @@ public void setStatus(AppointmentRequestStatus status) {
         this.status = status;
     }
 
+    public String sanitizeNotes(String note) {
+        return WebUtil.escapeHTML(note);
+    }
+
```
Comment (brandones) [inline api/src/main/java/org/openmrs/module/appointmentscheduling/AppointmentRequest.java:97]:
```suggestion
```

File: api/src/main/java/org/openmrs/module/appointmentscheduling/AppointmentRequest.java [hunk 3]
```diff
@@ -90,12 +91,16 @@ public void setStatus(AppointmentRequestStatus status) {
...
     public String getNotes() {
         return notes;
     }
 
     public void setNotes(String notes) {
-        this.notes = notes;
+        this.notes = sanitizeNotes(notes);
```
Comment (brandones) [inline api/src/main/java/org/openmrs/module/appointmentscheduling/AppointmentRequest.java:103]:
```suggestion
        this.notes = WebUtil.escapeHTML(notes);
```

**Comment by brandones** [comment]
This looks fine to me. @isears ?

**Review by mogoodrich — APPROVED** [review]

**Pull request merged**